bgpd/bgp_open.c

```diff
@@ -1896,7 +1896,8 @@ uint16_t bgp_open_capability(struct stream *s, struct peer *peer,
 	}
 
 	/* Hostname capability */
-	if (cmd_hostname_get()) {
+	if (CHECK_FLAG(peer->bgp->flags, BGP_FLAG_HOSTNAME_CAPABILITY) &&
```

Comment:
this flag should be on by default where are you setting it?

Reply:
because in the initial commit, I hesitated to change this behavior.
rationale : draft it not supported everywhere.

consequence if I disable this:
topotests may be complicated, especially with IPv6 setups where the link local nexthop is never tested.

Reply:
@donaldsharp this is enabled by default.

Reply:
boo on me.